go-controller/pkg/clustermanager/pod/allocator.go

```diff
@@ -190,11 +200,16 @@ func (a *PodAllocator) releasePodOnNAD(pod *corev1.Pod, nad string, network *net
 	hasPersistentIPs := network.IPAMClaimReference != ""
 	if hasPersistentIPs {
 		ipamClaim, err := a.findIPAMClaim(network)
-		if err != nil {
+		if err == nil {
+			hasPersistentIPs = ipamClaim != nil && len(ipamClaim.Status.IPs) > 0
+		} else if apierrors.IsNotFound(err) {
 			klog.Errorf("Failed to retrieve IPAMClaim %q but will release IPs: %v", network.IPAMClaimReference, err)
 			hasPersistentIPs = false
 		} else {
-			hasPersistentIPs = ipamClaim != nil && len(ipamClaim.Status.IPs) > 0
+			return fmt.Errorf(
+				"failed to find IPAMClaim %q",
+				fmt.Sprintf("%s/%s", network.Namespace, network.IPAMClaimReference),
+			)
```

Comment:
This works but I think it might be just nicer to follow a similar logic and naming as the allocation, so something like
```
hasPersistentIPs := netInfo.AllowsPersistentIPs() && hasIPAM
hasIPAMClaim := network != nil && network.IPAMClaimReference != ""
if hasIPAMClaim && !hasPersistentIPs {
    klog.Errorf(
        "Pod %s/%s referencing an IPAMClaim on network %q which does not honor it",
	pod.GetNamespace(),
	pod.GetName(),
	netInfo.GetNetworkName(),
    )
    hasIPAMClaim = false
}
if hasIPAMClaim {
   ipamClaim, err = a.ipamClaimsReconciler.FindIPAMClaim(network.IPAMClaimReference, network.Namespace)
   hasIPAMClaim = ipamClaim != nil && len(ipamClaim.Status.IPs) > 0
   if apierrors.IsNotFound(err)  {
      klog.Errorf("Failed to retrieve IPAMClaim %q but will release IPs: %v", network.IPAMClaimReference, err)
   } else if err != nil {
      return fmt.Errorf("failed to find IPAMClaim %s/%s: %w", network.Namespace, network.IPAMClaimReference, err)
   }
}
```

Reply:
Done.